test/NuGet.Core.Tests/NuGet.Packaging.Test/UpholdBuildConventionRuleTests.cs

```diff
@@ -0,0 +1,222 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Castle.DynamicProxy.Generators.Emitters;
+using NuGet.Common;
+using NuGet.Packaging.Rules;
+using Org.BouncyCastle.Asn1.X509;
+using Xunit;
+
+namespace NuGet.Packaging.Test
+{
+    public class UpholdBuildConventionRuleTests
```

Comment:
Here are some ideas for more tests:

* `build/extra/packageId.props` should warn because there is no `build/packageId.props`.
* `build/extra/packageId.props` with a `build/packageId.props` should not warn.
* `build/net45/extra/packageId.props` should warn because there is no `build/net45/packageId.props`.
* `build/net45/extra/packageId.props` with a `build/net45/packageId.props` should not warn.
* `build/random.file` should not warn (it's neither props nor targets)